pkg/php/composer/parse.go

```diff
@@ -24,18 +34,75 @@ func NewParser() types.Parser {
 
 func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
 	var lockFile lockFile
-	decoder := json.NewDecoder(r)
-	err := decoder.Decode(&lockFile)
+	input, err := io.ReadAll(r)
 	if err != nil {
+		return nil, nil, xerrors.Errorf("read error: %w", err)
+	}
+	if err = jfather.Unmarshal(input, &lockFile); err != nil {
 		return nil, nil, xerrors.Errorf("decode error: %w", err)
 	}
 
-	var libs []types.Library
+	libs := map[string]types.Library{}
+	foundDeps := map[string][]string{}
 	for _, pkg := range lockFile.Packages {
-		libs = append(libs, types.Library{
-			Name:    pkg.Name,
-			Version: pkg.Version,
+		lib := types.Library{
+			ID:        utils.PackageID(pkg.Name, pkg.Version),
+			Name:      pkg.Name,
+			Version:   pkg.Version,
+			Indirect:  false, // composer.lock file doesn't have info about Direct/Indirect deps. Will think that all dependencies are Direct
+			License:   strings.Join(pkg.License, ", "),
+			Locations: []types.Location{{StartLine: pkg.StartLine, EndLine: pkg.EndLine}},
+		}
+		libs[lib.Name] = lib
+
+		var dependsOn []string
+		for depName := range pkg.Require {
+			// Require field includes required php version, skip this
+			// Also skip PHP extensions
+			if depName != "php" || strings.HasPrefix(depName, "ext") {
+				dependsOn = append(dependsOn, depName) // field uses range of versions, so later we will fill in the versions from the libraries
+			}
+		}
+		if len(dependsOn) > 0 {
+			foundDeps[lib.ID] = dependsOn
+		}
+	}
+
+	// fill deps versions
+	var deps []types.Dependency
+	for libID, depsOn := range foundDeps {
+		var dependsOn []string
+		for _, depName := range depsOn {
+			if lib, ok := libs[depName]; ok {
+				dependsOn = append(dependsOn, lib.ID)
+				continue
+			}
+			log.Logger.Debugf("unable to find version of %s", depName)
+		}
+		dependsOn = sortDependsOn(dependsOn)
+		deps = append(deps, types.Dependency{
+			ID:        libID,
+			DependsOn: dependsOn,
 		})
 	}
-	return libs, nil, nil
+
+	return maps.Values(libs), deps, nil
```

Comment:
Do we also want to sort libs?

Reply:
it makes sense.
I have moved the sort functions to the `utils` package. So, we can use these functions in other packages.

Reply:
It is better to use sort.Interface. I've updated in https://github.com/aquasecurity/go-dep-parser/pull/197/commits/875540ff0617f5dc56f6567ef13dcde4c22412f5